### DISCUSSION TITLE
Service Dependency Graph Panel

### DISCUSSION THREAD

**Discussion by klayders** [grafana/grafana#47513] [Observability]
Hello, can we see the possibility of using tempo in grafan to display microservice communication as a graph?
Like this https://grafana.com/grafana/plugins/novatec-sdg-panel/

**Comment by connorlindsey** [comment]
We are currently working on this functionality behind a feature toggle. The docs for enabling it in the agent are here: https://grafana.com/docs/tempo/next/grafana-agent/service-graphs/

And docs for the UI are here: https://grafana.com/docs/grafana/latest/datasources/tempo/#service-graph. It requires the `tempoServiceGraph` feature toggle to appear in the UI

**Comment by nicolastakashi** [comment]
Hi @connorlindsey!

Is this feature also available for Jaeger?

**Comment by connorlindsey** [comment]
@nicolastakashi Not at the moment. The underlying Node Graph panel is datasource-agnostic, but the data generated to populate the panel is done specifically in Tempo. As of Tempo 1.4, the preferred way to get service graph data is by running the [metrics generator](https://grafana.com/blog/2022/05/02/new-in-grafana-tempo-1.4-introducing-the-metrics-generator/) in Tempo

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
